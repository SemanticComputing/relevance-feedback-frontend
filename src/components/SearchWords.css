.remove {
  margin-right: 5px;
}
