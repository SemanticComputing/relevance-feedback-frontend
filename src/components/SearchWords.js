import React from 'react';
import './SearchWords.css';
import { array, func } from 'prop-types';
import FontAwesome from 'react-fontawesome';
import { Button } from 'reactstrap';
import { map, flatten, uniq } from 'lodash';

const SearchWords = ({ words, removeWord }) => {

  return map(uniq(flatten(words)), (word) => (
    <Button className="remove" name="remove" key={word} onClick={() => removeWord(word)}>{word}
      {' '}<FontAwesome name="times-circle" />
    </Button>
  ));
};

SearchWords.propTypes = {
  words: array,
  removeWord: func
};

export default SearchWords;
