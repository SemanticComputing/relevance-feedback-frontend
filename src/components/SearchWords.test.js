import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import SearchWords from './SearchWords';

const removeWord = jest.fn();
const words = [
  'innovation',
  'technology',
  'something',
  'trash'
];

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<SearchWords removeWord={removeWord} words={words} />, div);
});

it('renders the search words', () => {
  const tree = renderer
    .create(<SearchWords removeWord={removeWord} words={words} />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});

it('calls removeWord when a word is clicked', () => {
  const resultElem = mount(<SearchWords removeWord={removeWord} words={words} />);
  const buttons = resultElem.children({ name: 'remove' });

  expect(buttons.length).toBe(4);

  buttons.at(0).simulate('click');

  expect(removeWord.mock.calls.length).toBe(1);
  expect(removeWord.mock.calls[0][0]).toEqual('innovation');

  buttons.at(2).simulate('click');

  expect(removeWord.mock.calls.length).toBe(2);
  expect(removeWord.mock.calls[1][0]).toEqual('something');
});
