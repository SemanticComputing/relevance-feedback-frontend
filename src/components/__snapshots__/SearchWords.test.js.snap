// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the search words 1`] = `
Array [
  <button
    className="remove btn btn-secondary"
    name="remove"
    onClick={[Function]}
    type="button"
>
    innovation
     
    <span
        aria-hidden={true}
        className="fa fa-times-circle"
    />
</button>,
  <button
    className="remove btn btn-secondary"
    name="remove"
    onClick={[Function]}
    type="button"
>
    technology
     
    <span
        aria-hidden={true}
        className="fa fa-times-circle"
    />
</button>,
  <button
    className="remove btn btn-secondary"
    name="remove"
    onClick={[Function]}
    type="button"
>
    something
     
    <span
        aria-hidden={true}
        className="fa fa-times-circle"
    />
</button>,
  <button
    className="remove btn btn-secondary"
    name="remove"
    onClick={[Function]}
    type="button"
>
    trash
     
    <span
        aria-hidden={true}
        className="fa fa-times-circle"
    />
</button>,
]
`;
